python_terraform/terraform.py

```diff
@@ -431,34 +408,38 @@ def set_workspace(self, workspace, *args, **kwargs) -> CommandOutput:
         :param workspace: the desired workspace.
         :return: status
         """
-        return self.cmd("workspace", "select", workspace, *args, **kwargs)
+        global_opts = self._generate_default_general_options(False)
+        return self.cmd(global_opts, "workspace", "select", workspace, *args, **kwargs)
 
     def create_workspace(self, workspace, *args, **kwargs) -> CommandOutput:
         """Create workspace
 
         :param workspace: the desired workspace.
         :return: status
         """
-        return self.cmd("workspace", "new", workspace, *args, **kwargs)
+        global_opts = self._generate_default_general_options(False)
+        return self.cmd(global_opts, "workspace", "new", workspace, *args, **kwargs)
 
     def delete_workspace(self, workspace, *args, **kwargs) -> CommandOutput:
         """Delete workspace
 
         :param workspace: the desired workspace.
         :return: status
         """
-        return self.cmd("workspace", "delete", workspace, *args, **kwargs)
+        global_opts = self._generate_default_general_options(False)
+        return self.cmd(global_opts, "workspace", "delete", workspace, *args, **kwargs)
 
     def show_workspace(self, **kwargs) -> CommandOutput:
         """Show workspace, this command does not need the [DIR] part
 
         :return: workspace
         """
-        return self.cmd("workspace", "show", **kwargs)
+        global_opts = self._generate_default_general_options(False)
+        return self.cmd(global_opts, "workspace", "show", **kwargs)
 
     def list_workspace(self) -> List[str]:
         """List of workspaces
-        
+
```

Comment:
Line 454 should be `(self.cmd(global_opts, "workspace", "list")[1] or '').split()`